packages/router/src/useBlocker.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useCallback, useState, useRef } from 'react'
+import { block, unblock, BlockerCallback } from './history'
+
+type BlockerState = 'IDLE' | 'BLOCKED'
+
+interface UseBlockerOptions {
+  when: boolean
+}
+
+export function useBlocker({ when }: UseBlockerOptions) {
+  const [blockerState, setBlockerState] = useState<BlockerState>('IDLE')
+  const [pendingNavigation, setPendingNavigation] = useState<(() => void) | null>(null)
+  const blockerId = useRef(`BLOCKER_${Date.now()}${Math.random()}`)
```

Comment:
I think you can use `useId()` to generate a ref id

Reply:
I'll check. This was meant to resemble. https://github.com/redwoodjs/redwood/blob/8ede10a586bb4873f6ff59f2a32c45d02090d04f/packages/router/src/history.tsx#L19